### PR TITLE
refactor: favourites repository

### PR DESCRIPTION
**Pull request by 7mochi**
# Describe your changes

## Checklist
- [x] I've manually tested my code

### CODE REVIEW

File: app/repositories/favourites.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+import app.state.services
+
+# +--------------+------------------------+------+-----+---------+----------------+
+# | Field        | Type                   | Null | Key | Default | Extra          |
+# +--------------+------------------------+------+-----+---------+----------------+
+# | userid       | int                    | NO   | PRI | NULL    |                |
+# | setid        | int                    | NO   |     | NULL    |                |
+# | created_at   | int                    | NO   |     | 0       |                |
+# +--------------+------------------------+------+-----+---------+----------------+
```
Comment (cmyui) [inline app/repositories/favourites.py:17]: UserID is the primary key? Doesn't this mean a user can only have a single favourite? Wtf

Comment (7mochi) [reply app/repositories/favourites.py:17]: i forgot setid :tf:

`primary key (userid, setid)`

Comment (cmyui) [reply app/repositories/favourites.py:17]: Kk, please use `desc` in the future to avoid any confusion/mis-reviews of PRs

Comment (cmyui) [reply app/repositories/favourites.py:17]: it slipped into the `create()` too -- the select isn't using the setid

Comment (cmyui) [reply app/repositories/favourites.py:17]: i'll fix -- check the diffs for some ideas perhaps

**Comment by cmyui** [comment]
tested & working

**Pull request merged**